api/dps/index.go

```diff
@@ -187,75 +207,42 @@ func (i *Index) Transaction(transactionID flow.Identifier) (*flow.Transaction, e
 	return &transaction, nil
 }
 
-// Transactions returns the transaction IDs within the given block.
-func (i *Index) Transactions(blockID flow.Identifier) ([]flow.Identifier, error) {
+// CollectionsForHeight returns the collection IDs within the given block.
+func (i *Index) CollectionsForHeight(height uint64) ([]*flow.LightCollection, error) {
```

Comment:
Do we need a slice of pointers here compared to a slice of structs?

Reply:
Whenever a struct represents a unique entity with a unique ID, it should be a pointer, in case it's changed anywhere. Otherwise, we would have out-of-date copies.